src/Microsoft.Extensions.FileProviders.Physical/PollingFileChangeToken.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Extensions.FileProviders.Physical
+{
+    public class PollingFileChangeToken : IChangeToken
+    {
+        private FileInfo _fileInfo;
+        private DateTime _previousWriteTimeUtc;
+        private DateTime _lastCheckedTimeUtc;
+        private bool _hasChanged;
+
+        public PollingFileChangeToken(FileInfo fileInfo)
+        {
+            _fileInfo = fileInfo;
+            _previousWriteTimeUtc = GetLastWriteTimeUtc();
```

Comment:
Is the assumption that the file exists when you create the object? There's a chance that the file could get deleted between the moment you get the FileInfo and the moment this code runs. If that happens, this change token will never fire because the last write time will always be DateTime.MinValue